File: codingame/onboarding.pl
#!/usr/bin/perl
use strict;
use warnings;
use feature 'say';
select(STDOUT); $| = 1;
for(;;){
	chomp(my $enemy_1 = <>); # name of enemy 1
	chomp(my $dist_1 = <>); # distance to enemy 1
	chomp(my $enemy_2 = <>); # name of enemy 2
	chomp(my $dist_2 = <>); # distance to enemy 2
	say($dist_1<$dist_2 ? $enemy_1 : $enemy_2);
}

File: codingame/onboarding.js
for(;;){
	var enemy1 = readline(); // name of enemy 1
	var dist1 = parseInt(readline()); // distance to enemy 1
	var enemy2 = readline(); // name of enemy 2
	var dist2 = parseInt(readline()); // distance to enemy 2
	print(dist1<dist2 ? enemy1 : enemy2);
}